### PR TITLE
OY2-17846 (Export All Page)

### PR DESCRIPTION
**Pull request by 13bfrancis**
## Purpose

The purpose of this Pull Request is to add a print screen that will allow the user to export all measures for a CoreSet to a PDF.

#### Linked Issues to Close

https://qmacbis.atlassian.net/browse/OY2-17846

## Approach

The end goal of this is to implement this page with PrinceXML (a service that allows for the generation of 508 Compliant PDFs). The wiring is in place for this, but the current implementation is a strictly frontend solution which works really well but does not implement 508 compliance. Further work will need to be done to implement the prince piece. In its current state with PrinceXML the pdf is not being styled properly.

#### Pull Request Creator Checklist

- [ ] This PR has an associated issue or issues.
- [ ] The associated issue(s) are linked above.
- [ ] This PR meets all acceptance criteria for those issues.
- [ ] This PR and linked issue(s) are adequately documented
- [ ] This PR and linked issues(s) are a complete description of the changeset; an individual or team should be able to understand the issue(s) and changes by reading through this PR and it's links, with no further interaction.
- [ ] Someone has been assigned this PR.
- [ ] At least one person has been marked as reviewer on this PR.

#### Pull Request Reviewer/Assignee Checklist

- [ ] This PR has an associated issue or issues.
- [ ] The associated issue(s) are linked above.
- [ ] This PR meets all acceptance criteria for those issues.
- [ ] This PR and linked issue(s) are adequately documented
- [ ] This PR and linked issues(s) are a complete description of the changeset; an individual or team should be able to understand the issue(s) and changes by reading through this PR and it's links, with no further interaction.

### CODE REVIEW

File: services/app-api/handlers/prince/pdf.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  const interceptor = aws4Interceptor({
+    region: "us-east-1",
+    service: "execute-api",
+  });
+
+  axios.interceptors.request.use(interceptor);
+
+  try {
+    const pdf = await axios.post(
+      "https://macpro-platform-dev.cms.gov/doc-conv/508html-to-508pdf",
```
Comment (benjaminpaige) [inline services/app-api/handlers/prince/pdf.ts:15]: should this be an env var?

File: services/ui-src/src/views/ExportAll/index.tsx [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+        document.styleSheets[i]?.cssRules[0]?.cssText.includes("--chakra") &&
+        document.styleSheets[i]?.cssRules[0]?.cssText.includes(":root")
+      ) {
+        const chakraVars = document.styleSheets[i];
+        document.body.setAttribute(
+          "style",
+          chakraVars.cssRules[0].cssText.split(/(\{|\})/g)[2]
+        );
+      }
+    }
```
Comment (benjaminpaige) [inline services/ui-src/src/views/ExportAll/index.tsx:41]: does this still need to happen if we are doing the thing on ln 44 below?

Comment (13bfrancis) [reply services/ui-src/src/views/ExportAll/index.tsx:41]: Unfortunately yes. What I am doing here is adding the chakra css properties/variables as inline styles for the body tag. I don't know why this is an issue, but unfortunately it is.

Comment (benjaminpaige) [reply services/ui-src/src/views/ExportAll/index.tsx:41]: got it. cool

**Review by karla-vm — APPROVED** [review]

**Pull request merged**